test.c

```diff
@@ -0,0 +1,134 @@
+#include "CurrentRange.h"
+#include "A2DConversion.h"
+#include <assert.h>
+
+int main()
+{
+int *chkrange; 
+int samplesarray[];
+ int arraysize;
+int *ptr; 
+ 
+ //Test case for current value range check
+printf("\n\nCurrentRange:Test case 1");
+samplesarray[] = {3, 3, 5, 4, 10, 11, 12};
+arraysize = sizeof(samplesarray) / sizeof(samplesarray[0]);
+chkrange = PrintRange(samplesarray,arraysize);
+//Test case to check start of range
+assert(*(chkrange+4) == 3);
```

Comment:
usage of 2d array for returning data is a good idea. Using structures would have improved readability.
But the test cases trying to perform pointer arithmetic for verification needs improvement. this kind of operations makes readability difficult and also decreases the effectiveness of test cases. Verification steps in test cases should be easily readable and understandable. 
The following code (though is programmatically correct) needs more time to understand. Scope for improvement!